### DISCUSSION TITLE
Trying to let Delta work together with Mercurial (hg)

### DISCUSSION THREAD

**Discussion by dahlia** [dandavison/delta#1045] [Ideas]
I tried to use Delta for [Mercurial] (`hg`) besides Git, and apparently it pretty works well!  The following [*.hgrc*][hgrc] settings enable Mercurial to work together with Mercurial:

```ini
[extensions]
hgext.extdiff =

[extdiff]
cmd.delta =

[alias]
diff = delta
```

Note that I declared `cmd.delta` using [`extdiff`][extdiff] first and then made an alias `hg diff` for `hg delta`.  At my first attempt, I did one-shot configuration instead like `cmd.diff = delta`, but it turned out that Mercurial repeated the below warning:

    extension 'extdiff' overrides commands: diff

While, aliasing to `hg diff` command warned nothing, so I settled in this workaround.

[Mercurial]: https://www.mercurial-scm.org/
[hgrc]: https://www.mercurial-scm.org/doc/hgrc.5.html
[extdiff]: https://www.mercurial-scm.org/wiki/RecordExtension

**Comment by paw-lu** [comment]
Haven't tried it yet, but the [docs suggest to add delta to the `[pager]` section of `.hgrc`](https://dandavison.github.io/delta/configuration.html?highlight=mercurial#git-config-file)

**Reply by ppetr** [reply]
Using `delta` as a pager doesn't work well, because then Mercurial uses it for all commands, not just `hg diff`. The solution based on `extdiff` seems to work much better. 👍
